bootstrapping-lambda/src/middleware/auth-middleware.ts

```diff
@@ -0,0 +1,35 @@
+import { getVerifiedUserEmail } from "../panDomainAuth";
+import { userHasPermission } from "../permissionCheck";
+import { NextFunction, Request, Response } from "express";
+import { HTTP_STATUS_CODES } from "../../../shared/http/httpClientValues";
+
+const MISSING_AUTH_COOKIE_MESSAGE =
+  "pan-domain auth cookie missing, invalid or expired";
+
+export interface AuthenticatedRequest extends Request {
+  userEmail?: string;
+}
+
+export const authMiddleware = async (
+  request: AuthenticatedRequest,
+  response: Response,
+  next: NextFunction
+) => {
```

Comment:
```suggestion
export const authMiddleware = (shouldStillCallNextForUnauthenticated?: true) => async (
  request: AuthenticatedRequest,
  response: Response,
  next: NextFunction
) => {
```
this might mean you could reuse this auth middleware for the `pinboard.loader.js` endpoint